core-primitives/test/src/mock/trusted_operation_pool_mock.rs

```diff
@@ -154,21 +151,22 @@ impl TrustedOperationPool for TrustedOperationPoolMock {
 		unimplemented!()
 	}
 
+	#[allow(clippy::type_complexity)]
 	fn ready(
 		&self,
 		shard: ShardIdentifier,
 	) -> Box<dyn Iterator<Item = Arc<Self::InPoolOperation>> + Send> {
 		let transactions = self.submitted_transactions.read().unwrap();
 		let ready_transactions = transactions
 			.get(&shard)
-			.map(|payload| payload.xts.iter().map(|xt| Self::map_stf_top_to_tx(xt)).collect())
-			.unwrap_or_else(|| vec![]);
+			.map(|payload| payload.xts.iter().map(Self::map_stf_top_to_tx).collect())
+			.unwrap_or_else(Vec::new);
```

Comment:
You can do `unwrap_or_default`

Reply:
yes good catch thanks 👍 have to make sure I remember that is these cases

Reply:
Although in this case, it would require use to explicitly specify the type on the returned variable (if we use `unwrap_or_default()`) - otherwise the type cannot be inferred. And specifying the type IMO is more code and less flexible?